precompiles/staking/staking.go

```diff
@@ -28,23 +28,28 @@ var _ vm.PrecompiledContract = &Precompile{}
 //go:embed abi.json
 var f embed.FS
 
+// PrecompileAddress defines the contract address of the staking precompile.
+const PrecompileAddress = "0x0000000000000000000000000000000000000800"
+
 // Precompile defines the precompiled contract for staking.
 type Precompile struct {
 	cmn.Precompile
 	stakingKeeper stakingkeeper.Keeper
 }
 
-// RequiredGas returns the required bare minimum gas to execute the precompile.
-func (p Precompile) RequiredGas(input []byte) uint64 {
-	methodID := input[:4]
-
-	method, err := p.MethodById(methodID)
+// LoadPrecompileABI loads the staking ABI from the embedded abi.json file
+// for the staking precompile.
+func LoadPrecompileABI() (abi.ABI, error) {
+	abiBz, err := f.ReadFile("abi.json")
 	if err != nil {
-		// This should never happen since this method is going to fail during Run
-		return 0
+		return abi.ABI{}, fmt.Errorf("error loading the staking ABI %s", err)
 	}
 
-	return p.Precompile.RequiredGas(input, p.IsTransaction(method.Name))
+	newAbi, err := abi.JSON(bytes.NewReader(abiBz))
+	if err != nil {
+		return abi.ABI{}, fmt.Errorf(cmn.ErrInvalidABI, err)
+	}
+	return newAbi, nil
 }
```

Comment:
move this to common directory

Reply:
Will do the common logic! There might still be the requirement to have a per precopmipile function to have the specific path to the abi.json